test/maps/azalias/zz_generated_alias_client.go

```diff
@@ -19,21 +19,23 @@ import (
 )
 
 type aliasClient struct {
-	endpoint string
-	pl       runtime.Pipeline
+	endpoint   string
+	apiVersion *string
+	pl         runtime.Pipeline
 }
 
 // newAliasClient creates a new instance of aliasClient with the specified values.
-func newAliasClient(geography *Geography, pl runtime.Pipeline) *aliasClient {
+func newAliasClient(geography *Geography, apiVersion *string, pl runtime.Pipeline) *aliasClient {
```

Comment:
I will say this is clunky as-is.  Since it behaves more like a boolean, I was thinking this should really look like `includeAPIVersion bool` instead.

Reply:
That does look better, so long as there's only one version. I wonder whether we really want a client parameter for this case though. If I understand correctly, in this case the swagger specifies one value for api-version but doesn't declare that param required. That implies the client needn't set api-version but if it does, there's only one valid value. It seems odd to make api-version configurable then--there's no reason to configure it. Should the client just use the known value?

Reply:
I agree that the behavior doesn't really make sense and is likely a bug in the swagger.  However, I don't think we want to have the code generator treat any params as "special" and bypass the authoring.  I believe some track 1 code generators did this with `api-version` which became problematic for some resource manager APIs that also have a parameter named `api-version`.

Reply:
Here's the description from the swagger which makes the intent a bit more clear

| "Specifies the version of the REST protocol used for processing the request. This is required when using shared key authorization."

So perhaps it's not a bug, however the API is a bit clunky (I suppose a hand-written convenience layer would simplify this).

This does make me realize that we aren't emitting the doc comments for global parameters though.

Reply:
I see that the swagger doesn't constrain `api-version` values, so strictly speaking we don't know that only one valid value exists (unless that's formally implied somehow; I don't know much about swagger syntax). Given that, I think we do want `*string` here.

It's my experience updating azblob's generated code that has me looking sideways at single-value parameters. Passing method parameters with only one acceptable value was awkward (I haven't checked whether the blob swagger constrains the parameters we generate "enums" for). I couldn't help but think, "there's only one value and you know what it is, why do I have to pass it to you" 😆

Reply:
For the linked service bus swagger, the value comes from the `default` clause.